packages/@aws-cdk/aws-ecs/test/ec2/integ.deployment-alarms.ts

```diff
@@ -0,0 +1,54 @@
+import * as autoscaling from '@aws-cdk/aws-autoscaling';
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as cdk from '@aws-cdk/core';
+import * as integ from '@aws-cdk/integ-tests';
+import * as cloudwatch from '@aws-cdk/aws-cloudwatch';
+import * as ecs from '../../lib';
+import { DeploymentControllerType } from '../../lib';
+
+const app = new cdk.App();
+const stack = new cdk.Stack(app, 'integ-deployment-alarms');
+
+const vpc = new ec2.Vpc(stack, 'Vpc', { maxAzs: 2 });
+
+const taskDefinition = new ecs.Ec2TaskDefinition(stack, 'TaskDef');
+
+taskDefinition.addContainer('web', {
+  image: ecs.ContainerImage.fromRegistry('amazon/amazon-ecs-sample'),
+  memoryReservationMiB: 256,
+});
+
+const autoScalingGroup = new autoscaling.AutoScalingGroup(stack, 'ASG', {
+  vpc,
+  instanceType: new ec2.InstanceType('t2.micro'),
+  machineImage: ecs.EcsOptimizedImage.amazonLinux2(),
+});
+
+const cp = new ecs.AsgCapacityProvider(stack, 'EC2CapacityProvider', {
+  autoScalingGroup,
+  // This is to allow cdk destroy to work; otherwise deletion will hang bc ASG cannot be deleted
+  enableManagedTerminationProtection: false,
+});
+
+const cluster = new ecs.Cluster(stack, 'EC2CPCluster', {
+  vpc,
+  enableFargateCapacityProviders: true,
+});
+
+cluster.addAsgCapacityProvider(cp);
+const myAlarm = cloudwatch.Alarm.fromAlarmArn(stack, 'myAlarm', 'arn:aws:cloudwatch:us-east-1:372902930120:alarm:myAlarm');
+
+new ecs.Ec2Service(stack, 'EC2Service', {
+  cluster,
+  taskDefinition,
+  deploymentController: {
```

Comment:
You don't need to manually specify this; the default deployment controller is `ECS`. But we should make sure to add validation logic and test for it in `ec2-service.test.ts` where we error out if [deployment controller](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ecs-service-deploymentcontroller.html) is "EXTERNAL" or "CODE_DEPLOY". 

This is true of `DeploymentCircuitBreaker` too (can't be enabled when DeploymentController is not ECS), so you can piggyback on any existing validation to also check DeploymentAlarms.

Reply:
Gotcha,
I added following validation with new test case for this. 
`if (props.deploymentAlarms
        && deploymentController
        && deploymentController.type !== DeploymentControllerType.ECS) {
      Annotations.of(this).addError('Deployment alarms requires the ECS deployment controller.');
    }`
    
   And
   also updated the integration file with latest snapshots